cmd/state-svc/internal/server/server.go

```diff
@@ -12,27 +13,35 @@ import (
 	"github.com/99designs/gqlgen/graphql/handler/transport"
 	"github.com/labstack/echo/v4"
 	"github.com/labstack/echo/v4/middleware"
+	"github.com/spf13/cast"
 
 	"github.com/ActiveState/cli/cmd/state-svc/internal/resolver"
 	genserver "github.com/ActiveState/cli/cmd/state-svc/internal/server/generated"
 	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
 )
 
 type Server struct {
+	cfg         *config.Instance
+	ctx         context.Context
+	done        chan struct{}
```

Comment:
nit: It might be worth noting that this channel tracks the reporting of doneness as opposed to the context done channel which indicates a request for doneness. It threw me off a bit reading through. Maybe rename to `wait`.